### PR TITLE
Add comments to composition demo

### PR DESCRIPTION
**Pull request by dhood**

### CODE REVIEW

File: composition/src/listener_component.cpp [hunk 1]
```diff
@@ -23,16 +23,31 @@
...
+ * Components get built into shared libraries and as such do not write their own main functions.
+ * The process using the component's shared library will instantiate the class as a ROS node.
+ */
 Listener::Listener()
 : Node("listener")
 {
+  /**
+    * Create a callback function for when messages are received.
+    * Variations of this function also exist using, for example, UniquePtr for zero-copy transport.
+    */
```
Comment (dirk-thomas) [inline composition/src/listener_component.cpp]: Should these be line comments rather then block comments. Doxygen (`/**`) would also not extract those I assume?

Comment (dirk-thomas) [inline composition/src/listener_component.cpp]: Same for the other files.

**Comment by wjwwood** [comment]
I updated this to use line comments in order to fix the inconsistent indentation and to avoid confusion with doxygen and to save vertical lines. I don't think we were planning on running doxygen on this anyways.

**Comment by dirk-thomas** [comment]
Still one file with block comments.

**Comment by wjwwood** [comment]
It's fixed.

**Comment by dhood** [comment]
Block comments are used in ROS 1 tutorials so I was following that
Thank you for iterating on this @wjwwood!

**Comment by wjwwood** [comment]
> Block comments are used in ROS 1 tutorials so I was following that

Ah, that makes sense. I was just updating it to follow our developer guide:

> Use /// and /** */ comments for documentation purposes and // style comments for notes and general comments
>
> -- https://github.com/ros2/ros2/wiki/Developer-Guide#c-1

Both paths were probably fine, but I switched it when touching things to fix the indentation issues.

**Pull request merged**